File: skill_sdk/tests/fixtures/cpp-project/src/shop/PercentageDiscount.cpp
#include "PercentageDiscount.h"

PercentageDiscount::PercentageDiscount(double percent) : percent_(percent) {}

double PercentageDiscount::Apply(const Order& order) {
    return order.RawTotal() * (1.0 - percent_ / 100.0);
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/discount/PercentageDiscount.java
package com.fixture.shop.discount;

import com.fixture.shop.models.Order;

/**
 * Applies a percentage off the raw total.
 */
public class PercentageDiscount implements DiscountStrategy {

    private final double percent;

    public PercentageDiscount(double percent) {
        this.percent = percent;
    }

    @Override
    public double apply(Order order) {
        return order.rawTotal() * (1.0 - percent / 100.0);
    }

    public double getPercent() {
        return percent;
    }
}
